src/PolykeyClient.ts

```diff
@@ -34,51 +50,66 @@ interface PolykeyClient extends CreateDestroyStartStop {}
 )
 class PolykeyClient {
   static async createPolykeyClient({
-    nodePath = config.defaultsUser.nodePath,
-    streamFactory,
-    streamKeepAliveTimeoutTime,
-    parserBufferByteLimit,
+    port,
+    host,
+    options = {},
     fs = require('fs'),
     logger = new Logger(this.name),
     fresh = false,
   }: {
-    nodePath?: string;
-    streamFactory: StreamFactory;
-    streamKeepAliveTimeoutTime?: number;
-    parserBufferByteLimit?: number;
+    port: number;
+    host: string;
+    options: DeepPartial<PolykeyClientOptions>;
     fs?: FileSystem;
     logger?: Logger;
     fresh?: boolean;
   }): Promise<PolykeyClient> {
     logger.info(`Creating ${this.name}`);
-    if (nodePath == null) {
+    const optionsDefaulted = utils.mergeObjects(options, {
+      nodePath: config.defaultsUser.nodePath,
+      tokenBase: config.paths.tokenBase,
+      callTimeoutTime: config.defaultsSystem.rpcCallTimeoutTime,
+      parserBufferSize: config.defaultsSystem.rpcParserBufferSize,
+    });
+
+    if (optionsDefaulted.nodePath == null) {
       throw new errors.ErrorUtilsNodePath();
     }
-    await utils.mkdirExists(fs, nodePath);
-    const sessionTokenPath = path.join(nodePath, config.paths.tokenBase);
+    await utils.mkdirExists(fs, optionsDefaulted.nodePath);
+    const sessionTokenPath = path.join(
+      optionsDefaulted.nodePath,
+      optionsDefaulted.tokenBase,
+    );
     const session = await Session.createSession({
       sessionTokenPath,
       logger: logger.getChild(Session.name),
       fresh,
     });
+    const ws = await WebSocketClient.createWebSocketClient({
+      host: host,
+      port: port,
+      config: optionsDefaulted.wsConfig,
+      logger: logger.getChild('WebSocketClient'),
+    });
     const rpcClientClient = new RPCClient({
       manifest: clientClientManifest,
-      streamFactory,
+      streamFactory: () => ws.connection.newStream(),
       middlewareFactory: rpcMiddleware.defaultClientMiddlewareWrapper(
         clientMiddleware.middlewareClient(session),
-        parserBufferByteLimit,
+        optionsDefaulted.parserBufferSize,
       ),
       toError: networkUtils.toError,
-      streamKeepAliveTimeoutTime,
+      streamKeepAliveTimeoutTime: optionsDefaulted.callTimeoutTime,
       logger: logger.getChild(RPCClient.name),
     });
     const pkClient = new this({
-      nodePath,
+      nodePath: optionsDefaulted.nodePath,
       rpcClientClient: rpcClientClient,
       session,
       fs,
       logger,
     });
+    pkClient.ws = ws;
```

Comment:
Don't do this, It should be provided to the constructor, not set after constructing.

```suggestion
```